lib/mongoose_push/telemetry/handler.ex

```diff
@@ -0,0 +1,42 @@
+defmodule MongoosePush.Telemetry.Handler do
+  @moduledoc """
+  Module responsible for handling telemetry events, in future should be integrated with Prometheus
+  """
+  use GenServer
+  require Logger
+
+  def start_link(_) do
```

Comment:
Is there any reason this module is a `GenServer`? This seems a bit too much considering it's use case is very simple, while adding async process here adds a lot complexity regarding back pressure issues (more events then target metric engine can handle), or simply slowing down event processing due to centralized nature of this `GenServer`.

Reply:
As long as we just update some metrics I agree that it's not needed.